app/helpers/where_helper.rb

```diff
@@ -0,0 +1,10 @@
+module WhereHelper
+  def where_or_stub_for(user)
+    user.last_whereabouts ||
+      OpenStruct.new(body: I18n.t('shruggie'), provenance: nil, sent_at: nil)
```

Comment:
Why not use an un-saved `WhereMessage` instance?

Reply:
😮 That hadn't occurred to me!